### ISSUE TITLE
Best way to compare old stocks with new ones

### ISSUE THREAD

**Issue by ae-mo** [robertmartin8/PyPortfolioOpt#270]
I would like to compare stocks that have been around for many years with others that are much newer. I was wondering what would be the best way to do this in PyPortfolioOpt.

As an hypothetical example we could imagine three stocks **A**, **B** and **C**. Let's assume **A** and **B** have price data starting from date **d1**, while **C** has price data starting from date **d2**, with **d2** > **d1**.

It certainly makes sense to optimize a portfolio consisting of **A** and **B**, using all their price data. But what if I wanted to add **C** to the portfolio? At which date should the dataframe start? If it started at **d1**, what would be the correct way to deal with missing **C** data?

EDIT: grammar

**Comment by robertmartin8** [comment]
Hi @ae-mo,

To properly optimise a portfolio (using mean-variance optimisation), you need to estimate expected returns and covariance. So the answer to your question is "do whatever gives you the best estimates" :) 

Most of the `expected_returns` and `risk_models` methods work fine with missing data – that is, the functions will ignore the missing data and compute what they can. For example, the `mean_historical_return` would compute the annualised expected returns using all available data of A,B,C, even though C had less data.

It's pretty much up to you as to whether this is sensible. My view is that it's fine unless there has been a major systematic shift in the markets that disproportionately benefits C. For example, if you have only 1 year of data for C and 10 years of data for A,B, but the last year was a huge bull market, then C will get a higher expected return than is fair. 

The alternative is to drop NaNs, i.e only use **d > d2** for all ABC. This is probably the more principled approach, provided that you have enough data. e.g if your most recent asset IPO'd in 2018, then you will still have 2y of data, so can optimise portfolios of up to ~50-100 assets. But if you want to include an asset that IPO'd last month (i.e less tha 30 data points), you won't be able to construct  proper covariance matrix.

TL;DR: Probably makes sense to use the same amount of data for each, as long as that gives you enough data. 

Hope this helps!
Robert
